core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -235,8 +236,9 @@ final class ShuffleBlockFetcherIterator(
   }
 
   private[this] def sendRequest(req: FetchRequest): Unit = {
+    val address = req.address.asInstanceOf[BlockManagerId]
```

Comment:
Could you add a simple assert before this?

Reply:
Sure!